pysrc/scripts/sampling.py

```diff
@@ -81,6 +99,41 @@ def clean_results(results_folder: str) -> None:
     for tmp_file in result_files:
         os.remove(os.path.join(os.getcwd(), results_folder, tmp_file))
 
+def get_file_combinations(samples: list[tuple[str, str, float]], config: GlobalConfiguration) \
+        -> list[tuple[str, str, float]]:
+    data_type_dict: dict[int, list[str]] = {}
+    for num_files_index in range(0, len(samples)):
+        for sample_file_index in range(0, len(samples[num_files_index])):
+
+            current_tuple = samples[num_files_index][sample_file_index]
+            path_to_data = current_tuple[0]
+            #TODO add handling of headers in files
+            df = pd.read_csv(path_to_data, sep=';', header=None, on_bad_lines='skip')
+            numeric_columns = df.select_dtypes(include=np.number).columns.tolist()
+            categorical_columns = df.select_dtypes(include='object').columns.tolist()
+
+            if len(numeric_columns) == 1:
+                dtype = "number"
+            elif len(categorical_columns) == 1:
+                dtype = "object"
+            else:
+                dtype = "other"
+
+            if dtype in data_type_dict.keys():
+                data_type_dict[dtype].append((num_files_index, sample_file_index))
+            else:
+                data_type_dict[dtype] = [(num_files_index, sample_file_index)]
+
+    datatype_tuples = []
+    for key in data_type_dict:
+        temp_list = []
+        for ele in range(0, len(data_type_dict[key])):
+            temp_list.append(samples[data_type_dict[key][ele][0]][data_type_dict[key][ele][1]])
+
+        datatype_tuples.append(temp_list)
+
+    return datatype_tuples
```

Comment:
My best guess for this method:

```suggestion
def get_file_combinations(samples: list[list[tuple[str, str, float]]], config: GlobalConfiguration) \
        -> list[list[tuple[str, str, float]]]:
    data_type_dict: dict[str, list[tuple[int, int]]] = {}
    for num_files_index in range(0, len(samples)):
        for sample_file_index in range(0, len(samples[num_files_index])):

            current_tuple = samples[num_files_index][sample_file_index]
            path_to_data = current_tuple[0]
            #TODO add handling of headers in files
            df = pd.read_csv(path_to_data, sep=';', header=None, on_bad_lines='skip')
            numeric_columns = df.select_dtypes(include=np.number).columns.tolist()
            categorical_columns = df.select_dtypes(include='object').columns.tolist()

            if len(numeric_columns) == 1:
                dtype = "number"
            elif len(categorical_columns) == 1:
                dtype = "object"
            else:
                dtype = "other"

            if dtype in data_type_dict.keys():
                data_type_dict[dtype].append((num_files_index, sample_file_index))
            else:
                data_type_dict[dtype] = [(num_files_index, sample_file_index)]

    datatype_tuples = []
    for columns_for_datatype in data_type_dict.values():
        temp_list = []
        for column_for_datatype in columns_for_datatype:
            temp_list.append(samples[column_for_datatype[0]][column_for_datatype[1]])

        datatype_tuples.append(temp_list)

    return datatype_tuples
```

I'm totally unsure whether I've understood it correctly, though.